### PR TITLE
feat: add connection edit

### PR DESCRIPTION
**Pull request by KonstantinDM**
Я не уверен, что это необходимо и правильно решено
Просто решил сделать, что нибудь простое, что бы познакомится с дартом и вашей концепцией =)

### CODE REVIEW

File: lib/pages/bucket_page/bucket_page.dart [hunk 1]
```diff
@@ -66,81 +95,47 @@ class BucketPage extends StatelessWidget {
           List<Object>? objects = state.items.objects;
           List<ListTile>? directoriesWidgets = [];
 
-          directories.forEach(
```
Comment (goshander) [inline lib/pages/bucket_page/bucket_page.dart:69]: a functional `forEach` is preferable if `for` is not explicitly required

File: lib/pages/bucket_page/bucket_page.dart [hunk 2]
```diff
@@ -66,81 +95,47 @@ class BucketPage extends StatelessWidget {
...
-                          severity: InfoBarSeverity.success,
-                        );
-                      });
-                    }
-                  },
-                ));
-              }
-            },
-          );
+          for (var object in objects) {
```
Comment (goshander) [inline lib/pages/bucket_page/bucket_page.dart]: up

File: lib/pages/bucket_page/bucket_page.dart [hunk 3]
```diff
@@ -41,10 +41,39 @@ class BucketPage extends StatelessWidget {
...
+        if (selectedDirectory != null) {
+          Minio minio = Minio(
+              endPoint: connection.endpoint,
+              accessKey: connection.accessKey,
+              secretKey: connection.secretKey);
+
+          await minio.fGetObject(bucket.name, object.key!,
+              path_lib.join(selectedDirectory, path_lib.basename(object.key!)));
+
+          if (!context.mounted) return;
```
Comment (goshander) [inline lib/pages/bucket_page/bucket_page.dart]: like for this

File: lib/pages/bucket_page/bucket_page.dart [hunk 4]
```diff
@@ -41,10 +41,39 @@ class BucketPage extends StatelessWidget {
           String object = state.path.isNotEmpty ? '$path/$fileName' : fileName;
 
           await minio.fPutObject(bucket.name, object, file.path);
+          if (!context.mounted) return;
```
Comment (goshander) [inline lib/pages/bucket_page/bucket_page.dart]: like for this

File: lib/pages/bucket_page/bucket_page.dart [hunk 5]
```diff
@@ -66,81 +95,47 @@ class BucketPage extends StatelessWidget {
...
-                        .read<BucketPageBloc>()
-                        .add(ObjectsRequested(prefix: newPath.join('/')));
-                  },
-                ));
-              }
-            },
-          );
+          for (var directory in directories) {
+            List<String> splittedPrefix = directory.split('/');
+            String? directoryName = splittedPrefix.length > 2
```
Comment (goshander) [inline lib/pages/bucket_page/bucket_page.dart:102]: can we use `path.dirname` or any built-in `path` lib method for this?

File: lib/pages/settings_page/settings_page.dart [hunk 6]
```diff
@@ -72,14 +87,40 @@ class SettingsPage extends StatelessWidget {
...
                   ),
-                  Button(
-                    child: const Text('Add connection'),
-                    onPressed: () async {
-                      context
-                          .read<SettingsPageBloc>()
-                          .add(const AddSubmitted());
-                    },
-                  ),
+                  edit
```
Comment (goshander) [inline lib/pages/settings_page/settings_page.dart]: maybe separate this code to the variable before return in `build` method

File: lib/widgets/settings_textbox.dart [hunk 7]
```diff
@@ -1,27 +1,21 @@
...
+      {super.key, this.label = '', this.value = '', required this.onChanged});
 
   final String label;
+  final String value;
   final Function(String) onChanged;
 
   @override
   Widget build(BuildContext context) {
-    return TextBox(
+    return Material(
```
Comment (goshander) [inline lib/widgets/settings_textbox.dart:13]: we use `fluent-ui` components, `Material` component is unexpected here

**Review by goshander — CHANGES_REQUESTED** [review]

File: lib/widgets/settings_textbox.dart (hunk 7 above)
Comment (KonstantinDM) [reply lib/widgets/settings_textbox.dart:13]: temporarily leave the Material

File: lib/pages/bucket_page/bucket_page.dart (hunk 5 above)
Comment (KonstantinDM) [reply lib/pages/bucket_page/bucket_page.dart:102]: I haven't touched this code. I'm not ready to decide how to do it right

File: lib/pages/bucket_page/bucket_page.dart (hunk 1 above)
Comment (KonstantinDM) [reply lib/pages/bucket_page/bucket_page.dart:69]: dart fix did that

File: lib/pages/bucket_page/bucket_page.dart (hunk 2 above)
Comment (KonstantinDM) [reply lib/pages/bucket_page/bucket_page.dart]: dart fix did that

File: lib/pages/bucket_page/bucket_page.dart (hunk 1 above)
Comment (KonstantinDM) [reply lib/pages/bucket_page/bucket_page.dart:69]: i reverted that decision and disabled that check

File: lib/pages/bucket_page/bucket_page.dart (hunk 2 above)
Comment (KonstantinDM) [reply lib/pages/bucket_page/bucket_page.dart]: i reverted that decision and disabled that check

**Pull request closed without merging**